astropy/wcs/wcsapi/high_level_api.py

```diff
@@ -351,26 +361,39 @@ def low_level_wcs(self):
         return self
 
     def world_to_pixel(self, *world_objects):
+        values = []
+        masks = []
+        for world_object in world_objects:
```

Comment:
The problem was that `SkyCoord.mask` always returns something, even if the coordinates themselves are not masked. Unfortunately, `Masked._get_data_and_masks` and `get_data_and_mask` do not take that into account, so we effectively had to reimplement those functions here. Longer term, I'd like to make those functions available in `astropy.utils.masked`, and then use those here. Options are that this gets merged first (and I open an issue to remind myself), or this waits for me to get around to it.